batch2/batch/driver/scheduler.py

```diff
@@ -19,6 +34,33 @@ def __init__(self, app):
         asyncio.ensure_future(self.loop('cancel_loop', self.cancel_state_changed, self.cancel_1))
         asyncio.ensure_future(self.bump_loop())
 
+    async def get_user_resources(self):
+        records = self.db.execute_and_fetchall(
+'''
+SELECT user, ready_cores_mcpu, running_cores_mcpu
+FROM user_resources
+WHERE ready_cores_mcpu > 0 OR running_cores_mcpu > 0;
+''')
+        return {record['user']: UserResources.from_record(record) for record in records}
+
+    async def compute_fair_share(self):
+        user_resources = await self.get_user_resources()
+        free_cores_mcpu = self.inst_pool.live_free_cores_mcpu
```

Comment:
live means pending or active.  healthy means active with not many connection errors.  We schedule on healthy nodes, so I think only healthy nodes should be considered here.